File: internal/extract/astgrep/testdata/integration/fixture.ts
// Minimal TypeScript fixture for per-rule-file integration tests.
// Covers: ts-func, ts-class, ts-interface, ts-type-alias, ts-enum, ts-method, ts-decorator,
//         ts-route-express (confirmed via ts-import-express signal),
//         ts-test-import-jest (test_import fact),
//         ts-test-fn (it()/test() call expressions → test_fn kind).

import express from 'express';
import { expect } from '@jest/globals';

export function greet(name: string): string {
    return `Hello, ${name}`;
}

export class Greeter {
    sayHello(): string {
        return "hello";
    }
}

export interface Salutation {
    message: string;
}

export type Greeting = string;

export enum Direction {
    Up,
    Down,
}

@Controller("/api")
export class ApiController {}

const app = express();
app.get('/hello', (req, res) => { res.send('hello'); });

// Exercises ts-test-fn: it()/test() call expressions emit test_fn facts.
// describe() is a suite, not a test, and must NOT match.
describe('greet', () => {
    it('greets by name', () => {
        expect(greet('world')).toBe('Hello, world');
    });
    test('handles empty name', () => {
        expect(greet('')).toBe('Hello, ');
    });
});

File: internal/extract/astgrep/testdata/integration/fixture.py
# Minimal Python fixture for per-rule-file integration tests.
# Covers: py-func, py-class, py-decorator,
#         py-route-decorator (confirmed via py-import-fastapi signal),
#         py-test-import-pytest (test_import fact),
#         py-lazy-import-module + py-lazy-import-from (lazy_import facts).

import pytest
from fastapi import FastAPI

app = FastAPI()


def process(data):
    """Public function — must appear in Syntax() facts."""
    return data


class Processor:
    """Public class."""
    pass


@staticmethod
def _hidden():
    """Private — must not appear in exported facts."""
    pass


@app.get('/items')
def list_items():
    """FastAPI route — confirmed by import above."""
    return []


# Lazy-import fixtures: in-function imports must produce lazy_import facts.
# The top-level `import pytest` and `from fastapi import FastAPI` above must NOT
# match (they are module-level, not inside a function body).
def lazy_loader():
    """Function with a deferred module import — must produce a lazy_import fact."""
    import os  # lazy: import_statement inside function_definition
    return os.getcwd()


def lazy_from_loader():
    """Function with a deferred from-import — must produce a lazy_import fact."""
    from pathlib import Path  # lazy: import_from_statement inside function_definition
    return Path(".")
